src/validation.cpp

```diff
@@ -474,8 +474,8 @@ class MemPoolAccept
         const bool m_test_accept;
     };
 
-    // Single transaction acceptance
-    bool AcceptSingleTransaction(const CTransactionRef& ptx, ATMPArgs& args) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+    // Single transaction acceptance, optionally takes an arguement to return tx fee to the caller
+    bool AcceptSingleTransaction(const CTransactionRef& ptx, ATMPArgs& args, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
I think the goal of `ATMPArgs` was that the function signatures don't ever need to change again. So I'd suggest to put the new arg there as well